core/src/main/java/org/elasticsearch/index/query/QueryBuilders.java

```diff
@@ -47,6 +47,14 @@ public static MatchAllQueryBuilder matchAllQuery() {
     }
 
     /**
+     * A query that match no documents.
+     */
+    // NO COMMIT: we should create a new MatchNoDocsQueryBuilder class for this instead?
+    public static EmptyQueryBuilder matchNoDocsQuery() {
+        return EmptyQueryBuilder.PROTOTYPE;
```

Comment:
I think we should not hack this around and we should avoid relying on something that might work now but might change in the future. let's add a MatchNoneQuery with a proper parser and builder?